### PR TITLE
Test against oracle

### PR DESCRIPTION
**Pull request by nickvergessen**
- [x] Based on #4735

### CODE REVIEW

**Comment by PVince81** [comment]
A nice start...
```
Nextcloud was successfully installed
An exception occurred while executing 'INSERT INTO "oc_talk_commands" ("app", "command", "name", "script", "response", "enabled") VALUES(?, ?, ?, ?, ?, ?)' with params ["", "help", "talk", "", 1, 3]:

ORA-01400: cannot insert NULL into ("AUTOTEST"."oc_talk_commands"."script")
```

**Comment by nickvergessen** [comment]
- [ ] Missing "update" step for the help command

File: lib/Migration/Version11000Date20201209142525.php [hunk 1]
```diff
@@ -103,7 +103,7 @@ public function changeSchema(IOutput $output, Closure $schemaClosure, array $opt
 			]);
 
 			$table->setPrimaryKey(['id']);
-			$table->addUniqueIndex(['session_hash'], 'tg_session_hash');
+			$table->addUniqueIndex(['session_hash'], 'tgn_session_hash');
```
Comment (jancborchardt) [inline lib/Migration/Version11000Date20201209142525.php:106]: This fixes my upgrade issue, so :+1: to that part for sure. :)

**Comment by nickvergessen** [comment]
Rebased to add "listable" column on the select

File: lib/Model/SelectHelper.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Model;
+
+use OCP\DB\QueryBuilder\IQueryBuilder;
+
+class SelectHelper {
+	public function selectRoomsTable(IQueryBuilder $query, string $alias = 'r'): void {
```
Comment (PVince81) [inline lib/Model/SelectHelper.php:29]: for readability on the calling side, I think I wouldn't make the alias optional.
I came here to check as I was confused where the alias comes from

File: lib/Model/SelectHelper.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+			->addSelect($alias . 'password')
+			->addSelect($alias . 'active_guests')
+			->addSelect($alias . 'active_since')
+			->addSelect($alias . 'last_activity')
+			->addSelect($alias . 'last_message')
+			->addSelect($alias . 'lobby_timer')
+			->addSelect($alias . 'object_type')
+			->addSelect($alias . 'object_id')
+			->addSelect($alias . 'listable')
+			->selectAlias($alias . 'id', 'r_id');
```
Comment (PVince81) [inline lib/Model/SelectHelper.php:51]: r_id should take the "r" from the alias, for consistency

File: lib/Manager.php [hunk 4]
```diff
@@ -241,9 +242,9 @@ public function loadLastCommentInfo(int $id): ?IComment {
 
 	public function resetAssignedSignalingServers(ICache $cache): void {
 		$query = $this->db->getQueryBuilder();
-		$query->select('*')
-			->from('talk_rooms')
-			->where($query->expr()->isNotNull('assigned_hpb'));
+		$query->select('r.*')
```
Comment (PVince81) [inline lib/Manager.php:245]: did you miss replacing this one with the helper ?

File: lib/Model/CommandMapper.php [hunk 5]
```diff
@@ -85,8 +85,13 @@ public function find(string $app, string $cmd): Command {
 		$query = $this->db->getQueryBuilder();
 		$query->select('*')
 			->from($this->getTableName())
-			->where($query->expr()->eq('app', $query->createNamedParameter($app)))
-			->andWhere($query->expr()->eq('command', $query->createNamedParameter($cmd)));
+			->where($query->expr()->eq('command', $query->createNamedParameter($cmd)));
```
Comment (PVince81) [inline lib/Model/CommandMapper.php:88]: should the select * above be replaced as well for Oracle ?

File: lib/Model/SelectHelper.php (hunk 2 above)
Comment (nickvergessen) [reply lib/Model/SelectHelper.php:29]: The alias is optional?

File: lib/Model/SelectHelper.php (hunk 3 above)
Comment (nickvergessen) [reply lib/Model/SelectHelper.php:51]: No, it needs to be `r_id` to match https://github.com/nextcloud/spreed/blob/70679077ca966f48e5dd710b75613b090742e915/lib/Manager.php#L178

File: lib/Manager.php (hunk 4 above)
Comment (nickvergessen) [reply lib/Manager.php:245]: yes, although in this case it's not a problem as we only request against 1 DB. but could do for consistency

File: lib/Model/CommandMapper.php (hunk 5 above)
Comment (nickvergessen) [reply lib/Model/CommandMapper.php:88]: No, it's only one table being queried, so it's fine.

File: lib/Model/SelectHelper.php (hunk 2 above)
Comment (nickvergessen) [reply lib/Model/SelectHelper.php:29]: Ah, you would **n't**.

Well it's r for room, a for attendee and s for session. Your IDE (if you have one) will also tell you that. Not a huge fan of adding hundreds of `'(a|r|s)` in the code just for lazy non-IDE users

Comment (PVince81) [reply lib/Model/SelectHelper.php:29]: it's not only for non-IDE but also when reading the code on Github for example, like when reviewing code changes or viewing a commit / diff on Github or CLI

but as you like, not a big issue to infer the alias

**Comment by nickvergessen** [comment]
Okay actually the problem was we first installed composer deps and then set the php version, so it ran composer with 8.0 before switching to 7.4
Nevertheless this led to the relevant fix in the dependency so 8.0 will work in the future.

**Review by PVince81 — APPROVED** [review]
:+1:

**Pull request merged**